ContentfulDeliveryAPI/CDARequestSerializer.m

```diff
@@ -8,32 +9,91 @@
 
 #import "CDARequestSerializer.h"
 #import "CDAUtilities.h"
+#import "Versions.h"
 
 @implementation CDARequestSerializer
 
--(instancetype)initWithAccessToken:(NSString*)accessToken {
+-(instancetype)initWithAccessToken:(NSString*)accessToken
+                      isCMARequest:(BOOL)isCMARequest {
     NSParameterAssert(accessToken);
 
     self = [super init];
     if (self) {
         [self setValue:[@"Bearer " stringByAppendingString:accessToken] forHTTPHeaderField:@"Authorization"];
 
-        self.userAgent = @"contentful.objc/2.0.0";
+        NSString *userAgentHeaderString = [self userAgentHeaderStringIsForCMA:isCMARequest];
+        [self setValue:userAgentHeaderString forHTTPHeaderField:@"X-Contentful-User-Agent"];
     }
     return self;
 }
 
--(void)setUserAgent:(NSString *)userAgent {
-    if (_userAgent == userAgent) {
-        return;
+- (NSString *)userAgentHeaderStringIsForCMA:(BOOL)isCMARequest {
```

Comment:
`userAgentHeaderStringIsForCMA` is confusingly named since it's also used for the CDA